src/otx/engine/engine.py

```diff
@@ -778,6 +780,100 @@ def explain(
+        Before the measurements, a warm-up is done.
+
+        Args:
+            checkpoint (PathLike | None, optional): Path to checkpoint. Optional for torch models. Defaults to None.
+            batch_size (int, optional): Batch size for benchmarking. Defaults to 1.
+            n_iters (int, optional): Number of iterations to average on. Defaults to 10.
+            extended_stats (bool, optional): Flag that enables printing of per module complexity for torch model.
+            Defaults to False.
+
+        Returns:
+            dict[str, str]: a dict with the benchmark results.
+        """
+        checkpoint = checkpoint if checkpoint is not None else self.checkpoint
+
+        if checkpoint is not None:
+            is_ir_ckpt = Path(checkpoint).suffix in [".xml"]
+            if is_ir_ckpt and not isinstance(self.model, OVModel):
+                # create OVModel
+                self.model = self._auto_configurator.get_ov_model(
+                    model_name=str(checkpoint),
+                    label_info=self.datamodule.label_info,
+                )
+
+            if not is_ir_ckpt:
+                model_cls = self.model.__class__
+                self.model = model_cls.load_from_checkpoint(
+                    checkpoint_path=checkpoint,
+                    map_location="cpu",
+                    **self.model.hparams,
+                )
+        elif isinstance(self.model, OVModel):
+            msg = "To run benchmark on OV model, checkpoint must be specified."
+            raise RuntimeError(msg)
+
+        self.model.eval()
+
+        def dummy_infer(model: OTXModel, batch_size: int = 1) -> float:
+            input_batch = model.get_dummy_input(batch_size)
+            start = time.perf_counter()
+            model.forward(input_batch)
+            end = time.perf_counter()
+            return end - start
+
+        warmup_iters = max(1, int(n_iters / 10))
+        for _ in range(warmup_iters):
+            dummy_infer(self.model, batch_size)
+
+        total_time = 0.0
+        for _ in range(n_iters):
+            total_time += dummy_infer(self.model, batch_size)
+        latency = total_time / n_iters
+        fps = batch_size / latency
+
+        final_stats = {"latency": f"{latency:.3f} s", "throughput": f"{(fps):.3f} FPS"}
+
+        if not isinstance(self.model, OVModel):
+            try:
+                from torch.utils.flop_counter import convert_num_with_suffix, get_suffix_str
+
+                input_batch = self.model.get_dummy_input(1)
+                model_fwd = lambda: self.model.forward(input_batch)
+                depth = 3 if extended_stats else 0
+                fwd_flops = measure_flops(self.model.model, model_fwd, print_stats_depth=depth)
+                flops_str = convert_num_with_suffix(fwd_flops, get_suffix_str(fwd_flops * 10**3))
+                final_stats["complexity"] = flops_str + " MACs"
+            except Exception as e:
```

Comment:
I think Flops is more common complexity indicator. Can we use it instead of MACs?

Reply:
In that case, we'd have to multiply the output of the flops counter by 2, because MACs are actually counted according to the source code, but this doesn't make sense, since accumulation and multiplication are effectively fused in modern hardware.
See https://www.kdnuggets.com/2023/06/calculate-computational-efficiency-deep-learning-models-flops-macs.html#:~:text=As%20mentioned%20in%20the%20previous,MACs%20equals%20roughly%20two%20FLOPs for instance